src/wallet/wallet2.cpp

```diff
@@ -355,7 +357,7 @@ std::unique_ptr<tools::wallet2> generate_from_json(const std::string& json_file,
       return false;
     }
 
-    wallet.reset(new tools::wallet2(testnet, restricted));
+    wallet.reset(make_basic(vm, opts).release());
```

Comment:
That's a good question. I think it's the JSON data that takes precedence, but not 100% sure. Not something the user ought to do with conflicting parameters anyway.